persimmon/website/views.py

```diff
@@ -133,3 +133,56 @@ def get_my_accounts(request):
         'balance': account.balance,
         'approval_status': account.approval_status,
     } for account in accounts]
+
+@api_function
+def get_my_address(request):
+    user = current_user(request)
+    return [{
+        'my address': user.address
+    }]
+
+@api_function
+def change_my_address(request, new_address: str):
+    user = current_user(request)
+    user.address = new_address
+    user.save()
+    return [{
+        'my new address': user.address
+    }]
+
+
+#transfer amount from the balance of bankaccount with acountnumb1 id
+@api_function
+def transfer_funds(request, accountnumb1: int, amount: Decimal, accountnumb2: int):
+    user = current_user(request)
+    account1 = BankAccount.objects.filter(owner=user, id=accountnumb1).exclude(approval_status=ApprovalStatus.DECLINED)
+    if len(account1)== 1:
+        if account1[0].balance < amount:
+            return [{
+                'error':'error insufficient funds'
+            }]
+        else:
+            account2= BankAccount.objects.filter(id=accountnumb2).exclude(approval_status=ApprovalStatus.DECLINED)
+            if len(account2)==1:
```

Comment:
A couple things to consider:

1) we need to add entries to the BankStatements table doing the accounting for the transaction
2) we want to be able to reuse code for the other kinds of logic that does actions against account balances

with that in mind maybe create a change_money function or something that does both the balance changing and the statement authoring?

additionally, we should also be thinking about the functionality we need to add for creating pending transfers between accounts owned by different users. When approved, those will run some logic that looks a lot like this. Maybe deal with this later via refactor?

Reply:
Very well, I'll make this function also create Bank statements and will write another function that will check how much the amount is first that will either make a pending transaction or call this function immediately